File: src/game/tools/rpc/output/wrap/DailyActivityDataWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class DailyActivityDataWrap : IWrapper
	{	
		public int ActivityIdx; // 数据下标
		public int ChestID; // 宝箱ID
		public int ActivityPoints; // 活跃度
		public int Lv; // 等级
		public int State; // 状态 0已经领取 1可以领取 2不可以领取
		
		public void Decode(Packet pck)
		{	
			ActivityIdx = pck.GetInt(); 
			ChestID = pck.GetInt(); 
			ActivityPoints = pck.GetInt(); 
			Lv = pck.GetInt(); 
			State = pck.GetInt(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutInt(ActivityIdx); 
        	pck.PutInt(ChestID); 
        	pck.PutInt(ActivityPoints); 
        	pck.PutInt(Lv); 
        	pck.PutInt(State); 
        }
	}
}

File: src/game/tools/rpc/output/wrap/DailyActivityDataWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type DailyActivityDataWrap struct {
	
	ActivityIdx int32 // 数据下标
	ChestID int32 // 宝箱ID
	ActivityPoints int32 // 活跃度
	Lv int32 // 等级
	State int32 // 状态 0已经领取 1可以领取 2不可以领取
	rpc.Wrapper
}

func (w *DailyActivityDataWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.ActivityIdx = pck.PopInt32(); 
	w.ChestID = pck.PopInt32(); 
	w.ActivityPoints = pck.PopInt32(); 
	w.Lv = pck.PopInt32(); 
	w.State = pck.PopInt32(); 
	return w
}

func (w *DailyActivityDataWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt32(w.ActivityIdx) 
	pck.PutInt32(w.ChestID) 
	pck.PutInt32(w.ActivityPoints) 
	pck.PutInt32(w.Lv) 
	pck.PutInt32(w.State) 
}

